Document Model Context Protocol Server OAuth with Home Assistant (#38279)
home-assistant/home-assistant.github.io: source/_integrations/mcp_server.markdown

## Client configuration

The Model Context Protocol specification has recently defined standards for
authorization and connecting to remote servers. The standards are a *work in progress*
and so some clients may not support the latest functionality, and the specification
will likely continue to evolve.

### Access control

#### OAuth

The Model Context Protocol supports OAuth for [Authorization](https://spec.modelcontextprotocol.io/specification/2025-03-26/basic/authorization/) and is fully supported by Home Assistant's
[Authentication API](https://developers.home-assistant.io/docs/auth_api/). MCP
Clients that support OAuth can use this to allow you to give the client access
to your Home Assistant MCP server.

> Home Assistant does not require you to pre-define an OAuth Client ID. Instead, the
> Client ID is simply the website of the application or the base of the redirect URL.

Reviewer: Should we mention IndieAuth here?

```suggestion
Home Assistant has adopted [IndieAuth](https://indieauth.spec.indieweb.org/) and does not require you to pre-define
an OAuth Client ID. Instead, the Client ID is the base of the redirect URL.
```

Reviewer: I also feel like we should be exact, and mention that it needs to be the base.

> - *Client ID*: This is the URL prefix such as `https://www.my-application.io`

Reviewer: ```suggestion
- *Client ID*: If your redirect-uri is `https://www.example.com/mcp/redirect`, your client ID should be `https://www.example.com`.
```
